src/lang/std/artifactGraph.ts

```diff
@@ -422,6 +422,54 @@ export function getArtifactsToUpdate({
       }
     })
     return returnArr
+  } else if (
```

Comment:
Below is the block of code that adds these new edge nodes to the graph, enabled by the opposite and adjacent requests made during execution `extrude.rs`.

Not this if statement is not easy to read, I would have liked to summarise them in variables so this just becomes `isOppositeEdge || isAdjacentEdge` but doing so means I loose the type narrowing in the if block, lmk if have ideas or ways to improve this.

Reply:
You can use type guards. https://www.typescriptlang.org/docs/handbook/advanced-types.html The purpose is exactly this: type narrowing based on data.

Reply:
Sorry still couldn't figure it out for `response` this was my best attempt but the type kept coming back as `never`

```ts
function isResponseOppositeEdge(
  response: Models['OkWebSocketResponseData_type']
): response is Extract<
Models['OkWebSocketResponseData_type'],
  {
    type: 'modeling'
    data: {
      modeling_response: {
        type: 'solid3d_get_opposite_edge'
        data: { edge: string }
      }
    }
  }
> {
  return (
    response.type === 'modeling' &&
    response.data !== undefined &&
    response.data.modeling_response !== undefined &&
    response.data.modeling_response.type === 'solid3d_get_opposite_edge' &&
    response.data.modeling_response.data !== undefined &&
    typeof response.data.modeling_response.data.edge === 'string'
  )
}
```

Reply:
Looks good to me as long as we've got something here